lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2095,6 +2095,50 @@ void SwiftLanguageRuntime::DidFinishExecutingUserExpression(
 
 bool SwiftLanguageRuntime::IsABIStable() { FORWARD(IsABIStable); }
 
+struct AsyncUnwindRegisters {
+  uint32_t async_ctx_regnum;
+  uint32_t fp_regnum;
+  uint32_t pc_regnum;
+  uint32_t dummy_regnum;
+};
+
+static llvm::Optional<AsyncUnwindRegisters>
+GetAsyncUnwindRegisters(llvm::Triple::ArchType triple) {
+  switch (triple) {
+  case llvm::Triple::x86_64:
+    return (AsyncUnwindRegisters){
+        .async_ctx_regnum = dwarf_r14_x86_64,
+        .fp_regnum = dwarf_rbp_x86_64,
+        .pc_regnum = dwarf_rip_x86_64,
```

Comment:
Why do we need to store pc_regnum here, doesn't LLDB already know this elsewhere?

Reply:
we have LLDB_GENERIC register numbers, but they use a different register number space.  The UnwindPlans need to be expressed in a single consistent register numbering, so we need the eh_frame / dwarf register numbers for pc/fp.  Normally you'd be right, we could use the generic_fp and generic_pc.

Reply:
(there is no LLDB_GENERIC that would get you the async context pointer.  You could argue for adding one, tbh, that's another way to approach this.)

Reply:
I captured some of this conversation in the doc comments.